packages/typespec-client-generator-core/src/internal-utils.ts

```diff
@@ -517,3 +525,55 @@ export function getAnyType(
     decorators: diagnostics.pipe(getTypeDecorators(context, type)),
   });
 }
+
+function pascalCaseForOperationId(name: string) {
+  return name
+    .split("_")
+    .map((s) => pascalCase(s))
+    .join("_");
+}
+
+/**
+ * Resolve the OpenAPI operation ID for the given operation using the following logic:
+ * - If @operationId was specified use that value
+ * - If operation is defined at the root or under the service namespace return `<operation.name>`
+ * - Otherwise(operation is under another namespace or interface) return `<namespace/interface.name>_<operation.name>`
+ *
+ * @param context TCGC context
+ * @param operation Operation
+ * @returns Operation ID in this format `<name>` or `<group>_<name>`
+ */
+export function resolveOperationId(context: TCGCContext, operation: Operation) {
```

Comment:
Any difference from `resolveOperationId` in typspec-openapi ?

Reply:
oh, forgot to removed.